### PR TITLE
Added Word Definitions feature and updated Readme. Fixes vaibhavsingh97/random-word#1

### PR DESCRIPTION
**Pull request by igagankalra**

### CODE REVIEW

**Comment by vaibhavsingh97** [comment]
@igagankalra Please resolve codacy issues and resolve conflicts.

**Comment by igagankalra** [comment]
@vaibhavsingh97  Why Travis CI check fails?

**Comment by vaibhavsingh97** [comment]
Ref: https://github.com/vaibhavsingh97/random-word/pull/4#issuecomment-402366739

File: random_word/tests/test_word.py [hunk 1]
```diff
@@ -12,9 +12,11 @@ def test_random_word(self):
     def test_random_words(self):
         return self.r.get_random_words()
 
-    def test_word_of_the_day(self):
-        return self.r.word_of_the_day()
-
+    def word_of_the_day(self):
+        return r.word_of_the_day()
+            
+    def word_meaning(self):
```
[the remark this answers is not in the document]
Comment (igagankalra) [reply random_word/tests/test_word.py]: @vaibhavsingh97  now who I can solve this, these are required there in the test file.

**Comment by igagankalra** [comment]
@vaibhavsingh97  I have updated the script according to the link you shared and tried every other possible thing through which I could solve Travis CI Check. Need some serious help here!!!

File: random_word/random_word.py [hunk 2]
```diff
@@ -134,8 +134,43 @@ def word_of_the_day(self, **kwargs):
...
+            partOfSpeech, str: CSV list of part-of-speech types (optional)
+            includeRelated, str: Return related words with definitions (optional)
+            sourceDictionaries, list: Source dictionary to return definitions from. (optional)
+            useCanonical, str: If true (default: false) will try to return the correct word root (optional)
+            includeTags, str: Return a closed set of XML tags in response (optional)
+
+        Returns: String, Definitions
+        """
+
+        url = "https://api.wordnik.com/v4/words.json/randomWord?"
```
Comment (vaibhavsingh97) [inline random_word/random_word.py]: Have you read the documentation? as this is not the calling endpoint.
Calling endpoint is `https://api.wordnik.com/v4/word.json/mistake/definitions?&api_key=YOURAPIKEY` without any parameters

File: random_word/random_word.py [hunk 3]
```diff
@@ -134,8 +134,43 @@ def word_of_the_day(self, **kwargs):
...
+        allParams = ['word', 'limit', 'partOfSpeech', 'includeRelated',
+                     'sourceDictionaries', 'useCanonical', 'includeTags']
+        params = locals()
+        payload = params['kwargs']
+        check_payload_items(payload, allParams)
+        payload['api_key'] = self.__api_key
+        del params['kwargs']
+        url += urlencode(payload, quote_via=quote_plus)
```
Comment (vaibhavsingh97) [inline random_word/random_word.py]: Also, verify the params too

File: .travis.yml [hunk 4]
```diff
@@ -9,4 +9,4 @@ script:
 - python random_word/tests/test_word.py
 before_install:
 - openssl aes-256-cbc -K $encrypted_619c3c835fae_key -iv $encrypted_619c3c835fae_iv
-  -in config.py.enc -out random_word/config.py -d
+  -in config.py.enc -out random_word/config.py -d
```
Comment (vaibhavsingh97) [inline .travis.yml]: Please don't change the file which is not related to PR. Add blank line

**Comment by vaibhavsingh97** [comment]
closing since there is no update. Feel free to open new PR
